.github/workflows/python-code-quality.yml

```diff
@@ -50,22 +51,25 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+          cache: pip
```

Comment:
Does this need to be part of this PR? Low risk?

Reply:
Kinda forgot to remove it, I've been adding it each time I work on my branches since when pip goes to install packages, the packages it wants to download and use are already present locally (downloaded in 1s at the beginning of the job).
It doesn't force pip to use any specific version (so no worries of not picking up an older dependency when it gets updated), it only already has the files available.
So, yes, low risk.

Reply:
Do you want it removed?

Reply:
You can keep it. I think it basically depends on how much confident you are that you won't need to revert rest of the PR: If we merge and see the main part does not work as expected, the cache improvement may get lost in the revert.

Reply:
no worries then